src/api/freeDictionaryAPI.ts

```diff
@@ -1,16 +1,23 @@
-import type { DefinitionProvider, DictionaryWord, Meaning, SynonymProvider } from "src/api/types";
-import type  DictionarySettings  from "src/types";
-
-import { LANGUAGES } from "src/_constants";
-import { debounce } from "obsidian";
-
-export default class FreeDictionaryAPI implements DefinitionProvider, SynonymProvider {
+import { DefinitionProvider, DictionaryWord, Meaning, PartOfSpeech, Synonym, SynonymProvider } from "src/api/types";
 
+class Base {
```

Comment:
I split the free dictionary api in two to make the APIs  little more consistent. Specifically, I added a PartOfSpeech provider, and didn't want to define `supportedLanguagesP`, as this could just keep growing. Now everything just uses `supportedLanguages`